src/rpp/rpp/operators/on_error_resume_next.hpp

```diff
@@ -0,0 +1,85 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2022 - present.
+//                    TC Wang 2022 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/defs.hpp>
+#include <rpp/operators/details/subscriber_with_state.hpp> // create_subscriber_with_state
+#include <rpp/operators/fwd/on_error_resume_next.hpp>
+#include <rpp/subscribers/constraints.hpp>
+
+IMPLEMENTATION_FILE(on_error_resume_next_tag);
+
+namespace rpp::details
+{
+
+template<rpp::details::resume_callable ResumeCallable>
+struct on_error_resume_state
+{
+    rpp::composite_subscription m_subscription;
+    ResumeCallable m_resume_callable;
+
+    explicit on_error_resume_state(rpp::composite_subscription subscription,
+                                   const ResumeCallable& callable) : m_subscription(std::move(subscription))
+                                                                   , m_resume_callable(callable) {};
+    explicit on_error_resume_state() = delete;
```

Comment:
default constructor automatically "deleted" in case of any manually specified constructor

Reply:
Oh interestingly. Where is that statement stated? In https://en.cppreference.com/w/cpp/language/default_constructor ?

Reply:
https://eel.is/c++draft/class.ctor#class.default.ctor-1.sentence-2

> If there is no user-declared constructor for class X, a non-explicit constructor having no parameters is implicitly declared as defaulted

```cpp
#include <optional>
#include <iostream>

struct NoCtr
{
    int v{};
};

struct Ctr
{
    Ctr(int v){}
    int v{};
};

int main()
{
    std::cout << "NoCtr " << std::is_default_constructible<NoCtr>() << std::endl;
    std::cout << "Ctr " << std::is_default_constructible<Ctr>() << std::endl;

    return 0;
}
```

```console
NoCtr 1
Ctr 0
```
[godbolt](https://godbolt.org/z/T5h6TEK7K)

Reply:
> Oh interestingly. Where is that statement stated? In https://en.cppreference.com/w/cpp/language/default_constructor ?


Your link also has something similar
> If no user-declared constructors of any kind are provided for a class type (struct, class, or union), the compiler will always declare a default constructor as an inline public member of its class.

> If some user-declared constructors are present, the user may still force the automatic generation of a default constructor by the compiler that would be implicitly-declared otherwise with the keyword default.

Reply:
So, it works in vice versa. Not deleted when has some user-defined ctor, but generated if no any user-defined